### PR TITLE
Cleanup of requirements and increase code quality

### PR DESCRIPTION
**Pull request by huppd**
This PR cleans up multiple things

- update miniconda scripts
- merge run and development requirements
- move jenkins testing to Balfrin
- activate pylint (cleanup errors and most warnings)

### CODE REVIEW

File: jenkins/mergerequest [hunk 1]
```diff
@@ -1,16 +1,15 @@
...
         }
     }
 
     stages {
         stage('Pytest') {
             steps {
                 sh'''
-                source /project/g110/icon/probtest/conda/miniconda/bin/activate
-                conda activate probtest
+                source $SCRATCH/../jenkins/project/g110/icon/probtest/conda/miniconda/bin/activate probtest
```
Comment (jonasjucker) [inline jenkins/mergerequest:12]: Isn't there a direct path to the env?
Why first go back with `../` to only go into `jenkins` again?

File: tests/conftest.py [hunk 2]
```diff
@@ -13,18 +23,18 @@
...
-    print(f"\nNew reference data will be stored in {new_ref}")
-    return new_ref
+    ref = os.path.join(tempfile.mkdtemp())
+    print(f"\nNew reference data will be stored in {ref}")
+    return ref
 
 
-@pytest.fixture()
-def ref_data() -> str:
+@pytest.fixture(name="ref_data")
```
Comment (jonasjucker) [inline tests/conftest.py:31]: nice trick :-)

File: util/icon/extract_timings.py [hunk 3]
```diff
@@ -160,6 +169,9 @@ def parse_time(time_string):
         h = 0
         m = 0
     else:
+        s = 0
+        m = 0
+        h = 0
         logger.error("did not match regex")
```
Comment (jonasjucker) [inline util/icon/extract_timings.py:175]: So far `logger.error` was always followed by `sys.exit(1)`.
Convert to logger.warning instead?

**Review by jonasjucker — APPROVED** [review]
Congrats to your little escalation.

Only some minor comments.

File: jenkins/mergerequest (hunk 1 above)
Comment (huppd) [reply jenkins/mergerequest]: I like to use the $SCRATCH variable her, as the absolute path has changed in the past on Balfrin. The `../` is such that it works for every user. But as it here is only meant to be the jenkins user, I am fine to remove it.

File: tests/conftest.py (hunk 2 above)
Comment (huppd) [reply tests/conftest.py:31]: :)

File: util/icon/extract_timings.py (hunk 3 above)
Comment (huppd) [reply util/icon/extract_timings.py]: I agree that makes sense.

**Review by jonasjucker — APPROVED** [review]
LGTM

**Pull request merged**